akka-projection-testkit/src/main/scala/akka/projection/testkit/TestProjection.scala

```diff
@@ -39,22 +43,58 @@ import akka.projection.scaladsl.SourceProvider
 @ApiMayChange
 object TestProjection {
+
+  /**
+   * Create a [[TestProjection]] that can be used to assert a [[akka.projection.scaladsl.Handler]] implementation.
+   *
+   * The [[TestProjection]] allows the user to test their [[akka.projection.scaladsl.Handler]] implementation in
+   * isolation, without requiring the Projection implementation (i.e. a database) to exist at test runtime.
+   *
+   * The [[akka.projection.scaladsl.SourceProvider]] can be a concrete implementation, or a [[TestSourceProvider]] to
+   * provide further test isolation.
+   *
+   * The [[TestProjection]] uses an at-least-once offset saving strategy where an offset is saved for each element.
+   *
+   * The [[TestProjection]] does not support grouping, at least once offset batching, or restart backoff strategies.
+   *
+   * @param projectionId - a Projection ID
+   * @param sourceProvider - a [[akka.projection.scaladsl.SourceProvider]] to supply envelopes to the Projection
+   * @param handler - a user-defined [[akka.projection.scaladsl.Handler]] to run within the Projection
+   */
   def apply[Offset, Envelope](
       projectionId: ProjectionId,
       sourceProvider: SourceProvider[Offset, Envelope],
       handler: () => Handler[Envelope]): Projection[Envelope] =
     new TestProjection(
-      projectionId,
-      sourceProvider,
-      SingleHandlerStrategy(handler),
-      AtLeastOnce(afterEnvelopes = Some(1)),
-      None)
+      projectionId = projectionId,
+      sourceProvider = sourceProvider,
+      handlerStrategy = SingleHandlerStrategy(handler),
+      // Disable batching so that `ProjectionTestKit.runWithTestSink` emits 1 `Done` per envelope.
+      offsetStrategy = AtLeastOnce(afterEnvelopes = Some(1)),
+      statusObserver = NoopStatusObserver,
+      offsetStoreFactory = system => TestInMemoryOffsetStore[Offset](system),
+      startOffset = None)
 
   /**
    * Java API
+   *
+   * Create a [[TestProjection]] that can be used to assert a [[akka.projection.javadsl.Handler]] implementation.
+   *
+   * The [[TestProjection]] allows the user to test their [[akka.projection.javadsl.Handler]] implementation in
+   * isolation, without requiring the Projection implementation (i.e. a database) to exist at test runtime.
+   *
+   * The [[akka.projection.javadsl.SourceProvider]] can be a concrete implementation, or a [[TestSourceProvider]] to
+   * provide further test isolation.
+   *
+   * The [[TestProjection]] uses an at-least-once offset saving strategy where an offset is saved for each element.
+   *
+   * The [[TestProjection]] does not support grouping, at least once offset batching, or restart backoff strategies.
+   *
+   * @param projectionId   - a Projection ID
+   * @param sourceProvider - a [[akka.projection.javadsl.SourceProvider]] to supply envelopes to the Projection
+   * @param handler        - a user-defined [[akka.projection.javadsl.Handler]] to run within the Projection
    */
   def create[Offset, Envelope](
```

Comment:
somewhat strange that we mix scaladsl and javadsl in the same here, but use different packages for ProjectionTestKit

Reply:
I was able to get away with overloads & alternative methods for most of the Java API, but to add the Java API for `TestInMemoryOffsetStore` I'll have to create different method names to support the different return values.

Do you mind if I split the API in #381?  I've added it to the TODO.